### PR TITLE
extractor/docker/docker.go: add parsing for tar.gz layers

### PR DESCRIPTION
**Pull request by R0GG**
ATM the function ExtractFromFile (https://github.com/knqyf263/fanal/blob/master/extractor/docker/docker.go#L259) does not support for layers in tar.gz format.

Kaniko generates images with this format:
```
-rw-r--r--  0 0      0        9229 Jan  1  1970 sha256:0e3e7470c618ac1a5cd6cc4266013a882e84783382ca629cc4b0cd02f641644f
-rw-r--r--  0 0      0    50380042 Jan  1  1970 5ae19949497e04289972756fe51cfac1a72b04fe2709e85a615945035c5a9a61.tar.gz
-rw-r--r--  0 0      0     7804488 Jan  1  1970 ed3d96a2798e8837be24597cabf44ce25585cb9db1d749299cb06d51349ea5c2.tar.gz
-rw-r--r--  0 0      0     9978163 Jan  1  1970 f1213685078145f6136360475dbaffd0f86dfe92133a7bc26d79602980b255dd.tar.gz
-rw-r--r--  0 0      0    51765388 Jan  1  1970 1a9ad5d5550bdff7db4c3d035bf9550bcd1de06a7f178a26de1d082591a5b956.tar.gz
-rw-r--r--  0 0      0   192264327 Jan  1  1970 6f18049a0455d5e717b580354b515d9ac661e1a28d6ca9d6f7bc85c0dd17a7cf.tar.gz
-rw-r--r--  0 0      0     5792282 Jan  1  1970 ce39fa9d79d10239f5407e0bf43e2056f10cab5386d7319e220478fbed6a8323.tar.gz
-rw-r--r--  0 0      0    26568003 Jan  1  1970 3a91ffcf88eadccd4b196138e3c783070af7e5239688128d2c7ed400289574e4.tar.gz
-rw-r--r--  0 0      0         235 Jan  1  1970 ee82cc8e15068a2e57dec1805ebc11ae639bf10d967bf1b859794756916bd0de.tar.gz
-rw-r--r--  0 0      0     1820807 Jan  1  1970 bf0dbf90a11587afe7b0b48c0f7bf182b2fb7823da8b298a45aa36977266369e.tar.gz
-rw-r--r--  0 0      0     2217699 Jan  1  1970 4527ef1bdba6c7ac88d2ead1e9999e390d81d4f1d6d48e549f34153369f38690.tar.gz
-rw-r--r--  0 0      0    92989718 Jan  1  1970 f53500454253b8d0d1d6ea6689b3d3bcf3ed5fdc27e2c8621578695cc89ec05f.tar.gz
-rw-r--r--  0 0      0     5039436 Jan  1  1970 aab07cb32e8b86745b8a0538c862a304509777f5970f302ccd33086bba80a709.tar.gz
-rw-r--r--  0 0      0        1080 Jan  1  1970 manifest.json
```

### CODE REVIEW

**Comment by knqyf263** [comment]
@R0GG Thanks! Would you fix the tests?

**Comment by R0GG** [comment]
@knqyf263 I updated the code and added a test

**Comment by R0GG** [comment]
Hello, @knqyf263 is there a problem with the PR?

**Comment by knqyf263** [comment]
@R0GG I'm sorry for the late reply. I has prepared the Contribution License Agreement (CLA). I'm planning to introduce automatic CLA bot, but would you sign manually for now?
https://portal.helloworks.com/link/wvhTF78a4A9hICrm

**Comment by R0GG** [comment]
@knqyf263 This CLA is for Trivy but the project is fanal, is it normal?

**Comment by knqyf263** [comment]
@R0GG I'm sorry. I added a sentence about `fanal`. If possible, I'm glad you sign the CLA.

**Comment by R0GG** [comment]
Sorry, for the late reply but now that you moved the repo to aquasecurity, is the CLA still valid?
btw is it possible to have a new updated link?

**Comment by knqyf263** [comment]
@R0GG I'm sorry for the late reply. You don't have to sign CLA now. I'll check it later.

File: extractor/docker/docker.go [hunk 1]
```diff
@@ -271,40 +274,68 @@ func (d DockerExtractor) ExtractFromFile(ctx context.Context, r io.Reader, filen
 		if err != nil {
 			return nil, extractor.ErrCouldNotExtract
 		}
+		tarFiles[header.Name], err = ioutil.ReadAll(tr)
```
Comment (knqyf263) [inline extractor/docker/docker.go]: I understand that you want to do this. But, if an image is too large, I think all content can't be stored in memory. What do you think?

File: extractor/docker/docker.go [hunk 2]
```diff
@@ -271,40 +274,68 @@ func (d DockerExtractor) ExtractFromFile(ctx context.Context, r io.Reader, filen
...
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	data, ok := tarFiles["manifest.json"]
+	if ok == false {
```
Comment (knqyf263) [inline extractor/docker/docker.go]:
```suggestion
	if !ok {
```

File: extractor/docker/docker.go [hunk 3]
```diff
@@ -271,40 +274,68 @@ func (d DockerExtractor) ExtractFromFile(ctx context.Context, r io.Reader, filen
...
+		return nil, err
+	}
+	if len(manifests) == 0 {
+		return nil, xerrors.New("Invalid manifest file")
+	}
+
+	// Extract the layers
+	for _, layerPath := range manifests[0].Layers {
+		data, ok := tarFiles[layerPath]
+		if ok == false {
```
Comment (knqyf263) [inline extractor/docker/docker.go]: Ditto

**Comment by knqyf263** [comment]
I merged it manually.

**Pull request closed without merging**